airgun/entities/host_new.py

```diff
@@ -32,6 +38,163 @@ def get_details(self, entity_name, widget_names=None):
         self.browser.plugin.ensure_page_safe()
         return view.read(widget_names=widget_names)
 
+    def edit_system_purpose(
+        self, entity_name, role=None, sla=None, usage=None, release_ver=None, add_ons=None
+    ):
+        """
+        Function that edits the system purpose of a host
+
+        Args:
+            entity_name: Name of the host
+            role: Role to be assigned
+            sla: SLA to be assigned
+            usage: Usage to be assigned
+            release_ver: Release version to be assigned
+            add_ons: Add-ons to be assigned
+
+        Raises:
+            ValueError: If no parameters are passed.
+        """
+
+        if not any([role, sla, usage, release_ver, add_ons]):
+            raise ValueError(
+                'At least one of the role, sla, usage, release_ver, add_ons ' 'must be provided!'
+            )
+        view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
+        view.wait_displayed()
+        view.overview.system_purpose.edit_system_purpose.click()
+        view = EditSystemPurposeView(self.browser)
```

Comment:
Why don't we use normal navigation here? I.e. `view = self.navigate_to(...` and something defined with `@navigator.register`?

Reply:
I have chosen this approach for implementing modal views like this because it is already implemented in the same way multiple times in `/view/host_new.py`. So I wanted to be consistent.

Reply:
And this paradigm is present in so many places already that I think it's not a blocker.

But the `navigator` paradigm is IMO at the core of airgun and should be followed where possible.

Reply:
Ok, I will try to take that in mind.